### ISSUE TITLE
Harmonize __repr__ for nodes

### ISSUE THREAD

**Issue by mpu-creare** [creare-com/podpac#297]
**Description**
Right now some of our nodes have a nice __repr__, but really we should have a default one

**Comment by jmilloy** [comment]
See also #341

**Comment by jmilloy** [comment]
**repr**

One line with class name, node type if needed, and key info.

```
>>> repr(sm)
<SMAP product='SPL4SMAU'>
>>> repr(node)
<Rasterio source='s3://bucket/path/to/file'>
```

**str**

One line with class name, node type if needed, key info, and a list of node attributes.

```
>>> sm
<SMAP product='SPL4SMAU' (attrs: base_url, interpolation, layer_key, output, product, units, version)>
>>> node
<Rasterio source='s3://bucket/path/to/file' (attrs: band, crs, interpolation, output, outputs, source, units)>
```

**Comment by jmilloy** [comment]
Note you can easily see the details using `json_pretty`:

```
>>> print(sm.json_pretty)                                                                                                        
{
    "SMAP_SPL4SMAU": {
        "plugin": "__main__",
        "node": "SMAP",
        "attrs": {
            "base_url": "https://n5eil02u.ecs.nsidc.org/opendap/SMAP",
            "interpolation": {
                "method": "nearest",
                "params": {
                    "time_tolerance": "2,h"
                }
            },
            "layer_key": "{rdk}sm_surface_analysis",
            "version": 4
        }
    }
}
>>> print(node.json_pretty)
{
    "Rasterio": {
        "node": "core.data.rasterio_source.Rasterio",
        "attrs": {
            "band": 1,
            "source": "s3://bucket/path/to/file"
        }
    }
}
```

But it is a bit obtuse. I wonder if we want something that will print output more like:

>>> node.attrs()
podpac.datalib.SMAP
	base_url: https://n5eil02u.ecs.nsidc.org/opendap/SMAP
	interpolation: {'method': 'nearest', 'params': {'time_tolerance': numpy.timedelta64(2,'h')}}
	layer_key: {rdk}sm_surface_analysis
	product: SPL4SMAU
	units: None
	version: 4
>>> 
podpac.core.data.rasterio_source.Rasterio'>
	band: 1
	crs: None
	interpolation: nearest
	output: None
	outputs: None
	source: s3://bucket/path/to/file
	units: None
```

**Comment by mpu-creare** [comment]
fixed by #371
